waku/v2/waku_discv5.nim

```diff
@@ -136,26 +130,51 @@ proc closeWait*(wd: WakuDiscoveryV5) {.async.} =
   wd.listening = false
   await wd.protocol.closeWait()
 
-proc findRandomPeers*(wd: WakuDiscoveryV5): Future[Result[seq[RemotePeerInfo], cstring]] {.async.} =
+proc findRandomPeers*(wd: WakuDiscoveryV5, pred: WakuDiscv5Predicate = nil): Future[seq[waku_enr.Record]] {.async.} =
```

Comment:
This `pred` is not set yet anywhere. In other words, the remaining work here is to add the static topic configuration and construct a predicate that checks ENRs against that. LMK if you planned to do this or if we should schedule it our side (no expectations, of course, just don't want to interfere if you have more PRs lined up :) )

Reply:
@LNSD, I apologise for not having seen this comment earlier. At some point I must have added it to a TODO and then lost track :)

For now I think the best idea is to use the `--topics` config option with the app then identifying this as a static shard configuration and (somehow) passing in the predicate for discv5 filtering. This indeed points to the fact that at least certain types of stand-beside discovery should happen as a service separate to the node. For now, depending on capacity, we may do something simpler first, but I agree that this is the direction we should be taking.

That said, you've helped us a lot so happy to take ownership to get this and the subsequent features merged. To make it a bit easier fort us to rebase (which may help fix some unrelated tests) I've merged your changes to a base repo branch in https://github.com/waku-org/nwaku/pull/1763. If you agree that we take over from here (thanks for getting this done!), we can close this PR without merging and continue the feature from there?